src/components/ItemDisplay.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useRef, useEffect } from 'react';
+import logo from '../assets/one.jpg';
+import { format } from 'date-fns';
+
+const ItemDisplay = ({ setShowModal, closeModal, details }) => {
```

Comment:
The same thing I mentioned for ListItem.jsx also applies here. If I imagine this to be a dialog component, it should internally handle it's state of open and close, and inform external clients/consumers when it's opened or closed through events. You can optionally add a `opened` props that tells it if it should be shown by default. Or by default show it as open, and add a prop to inform other components that it's closed. Similar code snippet I showed in `ListItem.jsx` could be adapted for this case.